File: go/domain/model/brew_method.go
package model

type BrewMethod string

var AllBrewMethods = []BrewMethod{PourOver, Espresso, FrenchPress, ColdBrew}

const (
	PourOver    BrewMethod = "POUR_OVER"
	Espresso    BrewMethod = "ESPRESSO"
	FrenchPress BrewMethod = "FRENCH_PRESS"
	ColdBrew    BrewMethod = "COLD_BREW"
)

File: typescript/src/domain/model/brew-method.ts
export const BrewMethod = {
  PourOver: "POUR_OVER",
  Espresso: "ESPRESSO",
  FrenchPress: "FRENCH_PRESS",
  ColdBrew: "COLD_BREW",
} as const;

export type BrewMethod = (typeof BrewMethod)[keyof typeof BrewMethod];

export const AllBrewMethods: BrewMethod[] = [
  BrewMethod.PourOver,
  BrewMethod.Espresso,
  BrewMethod.FrenchPress,
  BrewMethod.ColdBrew,
];
